### ISSUE TITLE
Low Recall

### ISSUE THREAD

**Issue by clydebailey** [AlexeyAB/darknet#1919]
I am training an XNOR version of tiny-yolov2 with 7th and 8th layer in XNOR. I found out some lines commented out 
133-136 from the convolutional_kernels.cu file. 
[https://github.com/AlexeyAB/darknet/blob/7f4b859514c7fcae1f9cc86276be31f133b5bc7c/src/convolutional_kernels.cu](url)
After uncommenting these lines, i trained for more than 100,000 iterations and got a decent precision but a bad a recall.

1. Why were those lines commented in the convolutional_kernel.cu file when we need the binarization input state?

2. Why am I getting a bad recall rate?
precision: 0.59
TP: 107
FP:73
FN:8793

**Comment by sporterman** [comment]
@clydebailey have you met the follow trouble?  hope some advice
![image](https://user-images.githubusercontent.com/13059352/48339310-e53ef900-e6a2-11e8-9e58-c82a1b23a60e.png)

**Comment by clydebailey** [comment]
@sporterman 
No. Your problems are related to your data files. Make sure that you have pointed your training data properly in obj.data and obj.names files

**Comment by sporterman** [comment]
@clydebailey  i'sure the two files are correct,  are  there any other paramsters  i need change

**Comment by clydebailey** [comment]
please show the .data and .names files along with the contents of training set.

**Comment by sporterman** [comment]
@clydebailey 
/database/yolov3/darknet1/data/obj.data:
![image](https://user-images.githubusercontent.com/13059352/48384417-3c85ad80-e725-11e8-897d-56ddb855a4ed.png)
/database/yolov3/darknet1/data/obj.names:
![image](https://user-images.githubusercontent.com/13059352/48384438-53c49b00-e725-11e8-8603-6454b54ce70a.png)
/database/yolov3/darknet1/data/train.txt:
![image](https://user-images.githubusercontent.com/13059352/48384506-b1f17e00-e725-11e8-8ac4-d163a05beb5e.png)
/database/yolov3/darknet1/data/obj:
![image](https://user-images.githubusercontent.com/13059352/48384554-eb29ee00-e725-11e8-8f6a-aa6f91938e09.png)
train command:   ./darknet detector train data/obj.data cfg/yolo-obj.cfg darknet53.conv74

**Comment by yongcong1415** [comment]
you dataset is coco? i only trained at pascal format,

**Comment by sporterman** [comment]
@yongcong1415 no. my personal dataset, now the obj is decreasing while training,,.....

**Comment by yongcong1415** [comment]
> @yongcong1415 no. my personal dataset, now the obj is decreasing while training,,.....

i mean you dataset format,you need use yolo_mark.exe to check you boxes position;

**Comment by clydebailey** [comment]
Try maintaining your images under "JPEG images" and annotations under "Annotations". This is the file format for Pascal training.

**Comment by sporterman** [comment]
@clydebailey how to use yolo_mark to check my data and label,
